### PR TITLE
Made get command better and added a fullBright command

### PR DESCRIPTION
**Pull request by nicobomber77**
-now if it's starving it will stop whatever it's doing and start getting food.
-added a new task to handle everything

-new fullBright command (can be helpful on stream)

usage: @ fullbright [float multiplier (default 1)]
default value is 1000.

### CODE REVIEW

File: src/main/java/adris/altoclef/commands/FullBrightCommand.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import adris.altoclef.commandsystem.ArgParser;
+import adris.altoclef.commandsystem.Command;
+import adris.altoclef.commandsystem.CommandException;
+
+public class FullBrightCommand extends Command
+{
+
+    public FullBrightCommand() throws CommandException
+    {
+        super("fullBright", "sets gamma really high", new Arg(String.class, "multiplier", "1", 0 ));
```
Comment (adrisj7) [inline src/main/java/adris/altoclef/commands/FullBrightCommand.java:18]: Instead of String.class I highly recommend using Float.class or Double.class, with the default value at 1.0 or something like that

Maybe a personal decision, but I'd do one of two things.
1) Rename this to "gamma" and have the float argument set the gamma directly
2) Keep this as fullbright but remove the multiplier value, as it might just make the command more confusing. If  you really want to keep it in or see it as useful, disregard this.

File: src/main/java/adris/altoclef/tasks/GetResourceTask.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
+package adris.altoclef.tasks;
```
Comment (adrisj7) [inline src/main/java/adris/altoclef/tasks/GetResourceTask.java:1]: This class is a good idea, but that's not exactly how this system handles situations like this.

The functionality of this class should be split into two parts:

1) A parameter for "isStarvingHungerThreshold" and "foodToCollectWhenStarving" (names may vary) in [Settings.java](https://github.com/adrisj7-AltoClef/altoclef/blob/main/src/main/java/adris/altoclef/Settings.java). By default this should be set to 0 or -1 or something to disable it
2) In [FoodChain.java](https://github.com/adrisj7-AltoClef/altoclef/blob/main/src/main/java/adris/altoclef/tasksystem/chains/FoodChain.java), at the very bottom of `getPriority`, check if we have less food units than the value in settings. If so, return a priority of something like 55 (will need to fix these magic numbers) and run `setTask(new CollectFoodTask)` with the parameter from settings.

After that, GetResourceTask shouldn't be needed anymore.

I would also consider when it's appropriate to need to get food. Is it when the player is starving/hungry, or when the player doesn't have enough food in its inventory?

File: src/main/java/adris/altoclef/commands/GetCommand.java [hunk 3]
```diff
@@ -25,7 +26,7 @@ protected void Call(AltoClef mod, ArgParser parser) throws CommandException {
         int count = parser.Get(Integer.class);
 
         if (TaskCatalogue.taskExists(resourceName)) {
-            Task targetTask = TaskCatalogue.getItemTask(resourceName, count);
+            Task targetTask = new GetResourceTask(resourceName, count);
```
Comment (adrisj7) [inline src/main/java/adris/altoclef/commands/GetCommand.java:29]: This change should be undone after you follow the next review comment. (Otherwise would be a valid solution)

**Review by adrisj7 — CHANGES_REQUESTED** [review]
FullBright's a great addition, though need to clarify a bit first.
Also your "GetResourceTask" is a nice idea, but needs to fit with the design of the project and needs to be configurable by the user in Settings. (this bot can do more than just get resources, this food collection should be global and not add extra bulkiness to each command)

File: src/main/java/adris/altoclef/commands/FullBrightCommand.java (hunk 1 above)
Comment (nicobomber77) [reply src/main/java/adris/altoclef/commands/FullBrightCommand.java:18]: Ok yeah i you're right, i will make it so it just sets the gamma, it's much easyer making it that way.

File: src/main/java/adris/altoclef/tasks/GetResourceTask.java (hunk 2 above)
Comment (nicobomber77) [reply src/main/java/adris/altoclef/tasks/GetResourceTask.java:1]: i'm actually still figuring out how the system works, so yes i'll do it this way.

Comment (nicobomber77) [reply src/main/java/adris/altoclef/tasks/GetResourceTask.java:1]: > A parameter for "isStarvingHungerThreshold" and "foodToCollectWhenStarving" (names may vary) in Settings.java.

I was already thinking to add the parameters in the settings but I didn't do it for some reason.

> In FoodChain.java, at the...

Ok, now I will do as you say. Before I didn't quite understand how the system worked so I did it this way and it worked, now I will change it.

**Comment by nicobomber77** [comment]
I was busy on the weekend but now I fixed everything and works well

Now I'm checking for food in the inventory since if it has food it will eat, so if it doesn't have much food in the inventory it goes to get some. It will go get more food than it needs to so it doesn't have to always go get it.

if it has less food than foodUnitsThreshold it goes to get foodUnitsToCollect amount of food.

by default is all disabled. good values I found work well are 10 for foodUnitsThreshold and 100 for foodUnitsToCollect.

**Comment by adrisj7** [comment]
Made a few changes+fixes after testing, you're welcome to take a look.

In FoodChain.java you removed a line that checked if we no longer had food, so the bot would get stuck if it was hungry and had no food. There were also a few minor problems (like a reference to "GetResourceTask"). Next time do a test or two on your machine if possible to compile, and maybe check for edge cases. Good stuff though!

Thanks for contributing!

**Comment by nicobomber77** [comment]
"It works on my machine!"... I forgot to stage the files in git lol
